```
Error running mint automation
+ mkdir -p 9450-8e4e865/gopath/src/github.com/minio
+ git -C 9450-8e4e865/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 9450-8e4e865/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 9450-8e4e865/gopath/src/github.com/minio/minio fetch --quiet minio pull/9450/head:pr9450
+ git -C 9450-8e4e865/gopath/src/github.com/minio/minio checkout --quiet pr9450
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/9450-8e4e865/gopath
+ make -C 9450-8e4e865/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/hdfs/v3 v3.0.1
go: downloading github.com/minio/gokrb5/v7 v7.2.5
go: downloading github.com/minio/minio-go/v6 v6.0.53
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading google.golang.org/api v0.5.0
go: downloading github.com/json-iterator/go v1.1.9
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/klauspost/compress v1.10.3
go: downloading github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: downloading github.com/tinylib/msgp v1.1.1
go: downloading github.com/cheggaaa/pb v1.0.28
go: downloading github.com/gorilla/handlers v1.4.0
go: downloading github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: downloading github.com/gorilla/rpc v1.2.0
go: downloading github.com/google/uuid v1.1.1
go: downloading github.com/minio/sio v0.2.0
go: downloading github.com/philhofer/fwd v1.0.0
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading github.com/montanaflynn/stats v0.5.0
go: downloading github.com/coreos/etcd v3.3.12+incompatible
go: downloading github.com/gorilla/mux v1.7.0
go: downloading github.com/prometheus/common v0.2.0
go: downloading github.com/Azure/azure-storage-blob-go v0.8.0
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading golang.org/x/sys v0.0.0-20200323222414-85ca7c5b95cd
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: downloading golang.org/x/crypto v0.0.0-20191117063200-497ca9f6d64f
go: downloading github.com/beevik/ntp v0.2.0
go: downloading github.com/coredns/coredns v1.4.0
go: downloading google.golang.org/grpc v1.22.0
go: downloading golang.org/x/net v0.0.0-20200324143707-d3edc9973b7e
go: downloading github.com/miekg/dns v1.1.8
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/golang/protobuf v1.3.2
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading go.uber.org/atomic v1.3.2
go: downloading github.com/mattn/go-colorable v0.1.1
go: downloading github.com/fatih/color v1.7.0
go: downloading github.com/Azure/azure-pipeline-go v0.2.1
go: downloading github.com/minio/lsync v1.0.1
go: downloading github.com/kurin/blazer v0.5.4-0.20200327014341-8f90a40f8af7
go: downloading github.com/mattn/go-isatty v0.0.7
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/secure-io/sio-go v0.3.0
go: downloading github.com/klauspost/reedsolomon v1.9.3
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/lib/pq v1.1.1
go: downloading github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/nats-io/nats.go v1.9.1
go: downloading github.com/klauspost/cpuid v1.2.2
go: downloading gopkg.in/ldap.v3 v3.0.3
go: downloading github.com/nats-io/jwt v0.3.2
go: downloading github.com/klauspost/pgzip v1.2.1
go: downloading github.com/hashicorp/vault/api v1.0.4
go: downloading github.com/pkg/errors v0.8.1
go: downloading github.com/minio/highwayhash v1.0.0
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/mattn/go-ieproxy v0.0.0-20190805055040-f9202b1cfdeb
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading github.com/Shopify/sarama v1.24.1
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/gogo/protobuf v1.2.1
go: downloading golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/hashicorp/vault/sdk v0.1.13
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/nats-io/nkeys v0.1.3
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading github.com/go-sql-driver/mysql v1.4.1
go: downloading github.com/nats-io/stan.go v0.4.5
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: downloading gopkg.in/ini.v1 v1.48.0
go: downloading github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: downloading google.golang.org/genproto v0.0.0-20190513181449-d00d292a067c
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.2.3
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pierrec/lz4 v2.4.0+incompatible
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/rs/cors v1.6.0
go: downloading gopkg.in/yaml.v2 v2.2.4
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: downloading golang.org/x/text v0.3.2
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: downloading cloud.google.com/go v0.39.0
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/minio/parquet-go v0.0.0-20200414234858-838cfa8aae61
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/eclipse/paho.mqtt.golang v1.2.0
go: downloading go.opencensus.io v0.21.0
go: downloading github.com/tidwall/gjson v1.3.5
go: downloading github.com/ryanuber/go-glob v1.0.0
go: downloading github.com/tidwall/sjson v1.0.4
go: downloading github.com/tidwall/match v1.0.1
go: downloading github.com/tidwall/pretty v1.0.0
go: downloading github.com/hashicorp/golang-lru v0.5.1
# github.com/minio/minio/cmd
cmd/xl-v1.go:186:38: undefined: d
make: *** [build] Error 2

```

--- a/cmd/xl-v1.go
+++ b/cmd/xl-v1.go
@@ -175,20 +175,21 @@ func getStorageInfo(disks []StorageAPI, endpoints Endpoints) StorageInfo {
 
 // StorageInfo - returns underlying storage statistics.
 func (xl xlObjects) StorageInfo(ctx context.Context, local bool) StorageInfo {
-	var endpoints = xl.endpoints
-	var disks []StorageAPI
 
-	if !local {
-		disks = xl.getDisks()
-	} else {
-		for i, d := range xl.getDisks() {
-			if endpoints[i].IsLocal && d.Hostname() == "" {
-				// Append this local disk since local flag is true
-				disks = append(disks, d)
+	disks := xl.getDisks()
+	if local {
+		var localDisks []StorageAPI
+		for i, disk := range disks {
+			if disk != nil {
+				if xl.endpoints[i].IsLocal && disk.Hostname() == "" {
+					// Append this local disk since local flag is true
+					localDisks = append(localDisks, disk)
+				}
 			}
 		}
+		disks = localDisks
 	}
-	return getStorageInfo(disks, endpoints)
+	return getStorageInfo(disks, xl.endpoints)
 }
 
 // GetMetrics - is not implemented and shouldn't be called.

